### PR TITLE
fix: Fix code errors reported by golangci-lint

### PR DESCRIPTION
**Pull request by pugangxa**
#### Motivation
In #53 reported some issues, checked and fixed the ones that are code errors

#### Modifications
Fix bug

#### Result
The code error is fixed, but
- The field alignment is not fixed, not sure whether it's needed and it may have potential impact, maybe just change the golangci-lint rule to ignore it?
- Does not upgrade golangci-lint to 1.42.1 and golang to 1.7 since as mentioned it will be looked separately.

### CODE REVIEW

**Comment by pugangxa** [comment]
Let me know if we want to address them all.

File: controllers/modelmesh/etcd.go [hunk 1]
```diff
@@ -39,11 +39,11 @@ func (m *Deployment) configureMMDeploymentForEtcdSecret(deployment *appsv1.Deplo
...
 
 			volumeMountExists := false
-			for _, volumeMount := range container.VolumeMounts {
+			for i, volumeMount := range container.VolumeMounts {
 				if volumeMount.Name == etcdVolume {
 					volumeMountExists = true
-					volumeMount.ReadOnly = true
-					volumeMount.MountPath = etcdMountPath
+					container.VolumeMounts[i].ReadOnly = true
+					container.VolumeMounts[i].MountPath = etcdMountPath
```
Comment (njhill) [inline controllers/modelmesh/etcd.go:46]: May as well avoid the struct copy altogether and also change to:
```go
			for i, _ := range container.VolumeMounts {
				if container.VolumeMounts[i].Name == etcdVolume {
```

**Review by njhill — COMMENTED** [review]
Thanks @pugangxa, and sorry I missed this earlier. These are legit bugs.

@pvaneck I agree with disabling the `fieldalignment` checks.

File: controllers/modelmesh/etcd.go (hunk 1 above)
Comment (pugangxa) [reply controllers/modelmesh/etcd.go]: OK, addressed.

**Review by njhill — APPROVED** [review]
Thanks @pugangxa!

For future reference, it's best not to modify existing commits that have already been reviewed (apart from rebasing them), just push new commits.

**Pull request merged**